### PR TITLE
Implement new settings UI

### PR DESCRIPTION
**Pull request by Tanza3D**

### CODE REVIEW

**Comment by EXtremeExploit** [comment]
IMO would be a lot nicer if when opening the settings ui to default to theme option :thinking:

**Comment by Tanza3D** [comment]
i wanted to do that but couldn't find any reasonable way to do it without rewriting alot, will follow up with another pr to do that in the future

**Comment by EXtremeExploit** [comment]
alrighty

**Review by EXtremeExploit — CHANGES_REQUESTED** [review]
format all js documents using vs code format document feature, there are parts where indent is completly wrong and is somewhat hard to keep up where scope ends and all of that, shortcut is Shift+Alt+F iirc, its just 2 files :)

File: global/js/navbar.js [hunk 1]
```diff
@@ -272,9 +264,241 @@ function markRead() {
...
+                setLanguage(key);
+            });
+        }
+    },
+    {
+        name: "medals",
+        icon: "oif-app-medals",
+        generate: async function (htmlInner) {
+            var section = settingUtils.genericSection();
+            settingUtils.linkedCheckbox("compiletely hide medals when unobtained filter enabled", "settings_medals__hidemedalswhenunobtainedfilteron", section, false, function (enabled) {
```
Comment (EXtremeExploit) [inline global/js/navbar.js:432]: typo

**Comment by Tanza3D** [comment]
the indents are apparently right @EXtremeExploit

**Review by EXtremeExploit — APPROVED** [review]
ok

**Pull request merged**